packages/server/src/utils/LocalFileSystemHandler.js

```diff
@@ -0,0 +1,124 @@
+
+const fs = require('fs')
+const path = require('path')
+const rimraf = require('rimraf');
+
+/**
+ * Store files in local file system (app-store/packages/server/upload)
+ */
+module.exports = class LocalFileSystemHandler {
+
+    constructor() {
+        this._uploadDirectory = __dirname + '/../../upload' //app-store/packages/server/upload
+        this.createDirectoryIfNotExists(this._uploadDirectory)
+    }
+
+    get directory() {
+        return this._uploadDirectory
+    }
+
+    createDirectoryIfNotExists(dir) {
+        if (!fs.existsSync(dir)) {
+            fs.mkdirSync(dir, { recursive: true, mode: '0744' });
+        }
+    }
+
+
+    saveFile(directoryPath, filename, buffer) {
+        return new Promise((resolve, reject) => {
+            try {
+                this.verifyPath(directoryPath)
+                this.createDirectoryIfNotExists(path.join(this._uploadDirectory, directoryPath))
+            } catch ( err ) {
+                reject(err)
+            }
+            
+            fs.writeFile(path.join(this._uploadDirectory, directoryPath, filename), buffer, (err) => {
```

Comment:
Afaik this code will still be executed in case of promise rejection, not 100% sure though.
If this is intended, nevermind :)